infra/Makefile.toml

```diff
@@ -0,0 +1,597 @@
+echo - Genesis file: ${GENESIS_FILE}
+echo
+"""
+
+[tasks.default]
+clear = true
+script_runner = "@duckscript"
+script = ['''
+    echo
+    echo Main tasks:
+    echo - clear: Clear all information: private keys, cache, database, etc.
+    echo - init: Initialize the system
+    echo - info: Print the setup information
+    echo - start: Start Fendermint/CometBFT docker containers with the genesis
+    echo - teardown: Stop the system and clears all data
+    echo - run: Init and then start the system
+    echo
+    echo Most tasks use these environment variables:
+    echo - CHAIN_NAME (default '${CHAIN_NAME}'): the target IPC subnet
+    echo
+    echo Run 'cargo make -e CHAIN_NAME=chain -e BALANCE=100 -e BASE_FEE=200 ... COMMAND' to populate the variables from CLI or
+    echo Run 'cargo make --env-file=/PATH/.env COMMAND' to populate the variables from the file before running the command.
+    echo
+    echo Run 'cargo make --list-all-steps' for a complete list of available tasks.
+    echo
+''']
+
+########################################################################################################################
+# Single node
+########################################################################################################################
+[tasks.node]
+workspace = false
+dependencies = [
+  "node-down",
+  "node-init",
+  "node-network-create",
+  "cometbft-init",
+  "fendermint-deps",
+  "node-config",
+  "fendermint-start",
+  "cometbft-start",
+  "cometbft-wait",
+  "ethapi-start"
+]
+
+[tasks.node-network-create]
+env = { "NETWORK_NAME"="${NETWORK_NAME}"}
+extend = "network-create"
+
+
+[tasks.node-init]
+dependencies = [
+  "node-clear",
+  "node-mkdir",
+]
+
+[tasks.node-clear]
+script="""
+echo clearing all IPC data
+rm -rf ${BASE_DIR}
+"""
+
+[tasks.node-mkdir]
+script="""
+echo creating directories: $BASE_DIR $FM_DIR $CMT_DIR
+mkdir -p $BASE_DIR
+mkdir -p $FM_DIR
+
+mkdir -p $CMT_DIR
+"""
+
+[tasks.node-down]
+# `dependencies` doesn't seem to work with `cleanup_task`.
+run_task = { name = [
+  "cometbft-stop",
+  "cometbft-rm",
+  "fendermint-stop",
+  "fendermint-rm",
+  "ethapi-stop",
+  "ethapi-rm",
+  "network-rm",
+]}
+
+[tasks.node-config]
+dependencies = [
+  "fendermint-newgenesis",
+  "fendermint-newkey",
+  "fendermint-newaccount",
+  "fendermint-newgateway",
+]
+
+[tasks.fendermint-newgenesis]
+cwd = "./target/release"
+script = """
+./fendermint genesis --genesis-file ${GENESIS_FILE} new --chain-name ${CHAIN_NAME} --base-fee ${BASE_FEE} --timestamp ${TIMESTAMP}
+"""
+
+[tasks.fendermint-newkey]
+cwd = "./target/release"
+script="""
+mkdir -p $KEYS_DIR
+
+./fendermint key gen --out-dir $KEYS_DIR --name $KEY_NAME;
+chmod 600 ${PRIV_KEY_PATH}
+"""
+
+[tasks.fendermint-newaccount]
+cwd = "./target/release"
+script="""
+./fendermint genesis --genesis-file ${GENESIS_FILE} add-account --public-key ${PUB_KEY_PATH} --balance ${BALANCE}
+"""
+
+
+[tasks.fendermint-newgateway]
```

Comment:
This task does a lot, it could use some docs. Why is gateway creation, which is just putting a section into the gensis file, coupled with adding a validator and exporting its key to CometBFT?

Reply:
I would even aggregate them all in one task because I do not see reasons to separate them in this Makefile, but I know what you mean, and I will divide them.
Fixed